server/series/api.py

```diff
@@ -0,0 +1,436 @@
+from server.schema import Response, TeamSchema
+from server.season.models import Season
+from server.types import message_response
+from server.utils import today
+
+from .models import Series, SeriesRegistration, SeriesRosterInvitation
+from .schema import (
+    AddOrRemoveTeamSeriesRegistrationSchema,
+    SeriesCreateSchema,
+    SeriesRegistrationAddSchema,
+    SeriesRegistrationSchema,
+    SeriesRosterInvitationCreateSchema,
+    SeriesRosterInvitationSchema,
+    SeriesSchema,
+    SeriesTeamRosterSchema,
+)
+from .utils import RegistrationError, register_player
+
+router = Router()
+
+
+class AuthenticatedHttpRequest(HttpRequest):
+    user: User
+
+
+@router.get("/all", auth=None, response={200: list[SeriesSchema]})
+def get_all_series(request: AuthenticatedHttpRequest) -> tuple[int, QuerySet[Series]]:
+    return 200, Series.objects.all()
+
+
+@router.get("/", auth=None, response={200: SeriesSchema, 400: Response})
+def get_series(
+    request: AuthenticatedHttpRequest, id: int | None = None, slug: str | None = None
+) -> tuple[int, Series | message_response]:
+    if id is None and slug is None:
+        return 400, {"message": "Need either id or slug to fetch series"}
+
+    try:
+        series = Series.objects.get(id=id) if id is not None else Series.objects.get(slug=slug)
+    except Series.DoesNotExist:
+        return 400, {"message": "Series does not exist"}
+
+    return 200, series
+
+
+@router.post("/", response={200: SeriesSchema, 400: Response, 401: Response})
+def create_series(
+    request: AuthenticatedHttpRequest, series_details: SeriesCreateSchema
+) -> tuple[int, Series | message_response]:
+    if not request.user.is_staff:
+        return 401, {"message": "Only Admins can create a series"}
+
+    series = Series(
+        start_date=series_details.start_date,
+        end_date=series_details.end_date,
+        name=series_details.name,
+        type=series_details.type,
+        category=series_details.category,
+        series_roster_max_players=series_details.series_roster_max_players,
+        event_min_players_male=series_details.event_min_players_male,
+        event_min_players_female=series_details.event_min_players_female,
+        event_max_players_male=series_details.event_max_players_male,
+        event_max_players_female=series_details.event_max_players_female,
+    )
+
+    if series_details.player_transfer_window_start_date:
+        series.player_transfer_window_start_date = series_details.player_transfer_window_start_date
+
+    if series_details.player_transfer_window_end_date:
+        series.player_transfer_window_end_date = series_details.player_transfer_window_end_date
+
+    if series_details.season_id:
+        try:
+            series.season = Season.objects.get(id=series_details.season_id)
+        except Season.DoesNotExist:
+            return 400, {"message": "Season does not exist"}
+
+    series.save()
+    return 200, series
+
+
+@router.put(
```

Comment:
api can be `POST /{series_slug}/team/{team_slug}`

Reply:
done